src/main/java/org/folio/marc4ld/service/marc2ld/Marc2BibframeMapperImpl.java

```diff
@@ -65,15 +66,30 @@ public Resource fromMarcJson(String marc) {
       return null;
     }
     var reader = getReader(marc);
-    var instance = new Resource().addType(INSTANCE);
+    var instanceAndWork = createInstanceAndWork();
```

Comment:
As we already know that an Instance and Work has to be created, create them upfront.